src/data/spotifyDAO.ts

```diff
@@ -16,21 +16,26 @@ export class SpotifyDAO {
             clientSecret,
         });
 
-        // Retrieve an access token.
+        this.fetchAccessToken();
+    }
+
+    private async fetchAccessToken() {
         this.api.clientCredentialsGrant().then(
             (data) => {
                 console.log('Spotify access token expires in ' + data.body.expires_in);
                 console.log('Spotify access token is ' + data.body.access_token);
-                setTimeout(() => {
-                    console.log('Spotify access token expired!');
-                }, data.body.expires_in * 1000);
 
                 // Save the access token so that it's used in future calls
                 this.api.setAccessToken(data.body.access_token);
+
+                // Recursively refetch after the duration of 'expires_in' - 1m buffer period
+                setTimeout(() => {
+                    this.fetchAccessToken();
+                }, data.body.expires_in * 1000 - 60000);
```

Comment:
when dealing with MS times, i usually try to be pretty explicit about numbers. not a big deal either way, but it'd be cool if this were:

`(data.body.expires_in - 60) * 1000`

Reply:
I like it! :+1: